src/statement.d

```diff
@@ -1324,38 +1329,38 @@ extern (C++) class CompoundStatement : Statement
         return statements;
     }
 
-    override final ReturnStatement isReturnStatement()
+    override final inout(ReturnStatement) isReturnStatement() inout nothrow pure
```

Comment:
I can imagine why the code was structured that way ... 
To have an easy determination if a block returns or not.
I do agree that it should not be structured that way.

However to be pragmatic would you accept this PR with a FIXME note ?